src/data_preprocessing.py

```
@@ -0,0 +1,213 @@
+
+        # 2. Processed Image (Inverted, Black background, 64x64)
+        processed_img = original_img.convert('L')
+        processed_img = ImageOps.invert(processed_img)  # <-- Invert for visualization too!
+        processed_img = processed_img.resize((IMG_SIZE, IMG_SIZE))
+
+        arr = np.array(processed_img).astype(np.float32) / 255.0
+        arr_to_save = (arr * 255).astype(np.uint8)
+        processed_save_path = os.path.join(output_vis_dir, 'processed_inverted_64x64.png')
+        Image.fromarray(arr_to_save).save(processed_save_path)
+
+        print(f"Visualization saved:")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)